### PR TITLE
[MIG] Port pos_pricelist to 9.0

### PR DESCRIPTION
**Pull request by x620**

### CODE REVIEW

File: pos_pricelist/views/point_of_sale_view.xml [hunk 1]
```diff
@@ -10,7 +10,18 @@
...
+                    <!--<page string="Taxes">-->
+                        <!--<field name="taxes">-->
+                            <!--<tree editable="bottom" string="Taxes detail">-->
+                                <!--<field name="name"/>-->
+                                <!--<field name="base"/>-->
+                                <!--<field name="amount"/>-->
+                            <!--</tree>-->
+                        <!--</field>-->
+                    <!--</page>-->
+                <!--</page>-->
```
Comment (legalsylvain) [inline pos_pricelist/views/point_of_sale_view.xml]: Hi,

Could your remove this commented useless part of code ?

File: pos_pricelist/demo/pos_pricelist_demo.yml [hunk 2]
```diff
@@ -62,34 +62,35 @@
...
         sequence: 1
         min_quantity: 2
         product_id: pos_product_product_3
-        base: -2
+#        base: -2
       - name: 5% Discount on all Computer related products (Qty 2)
         sequence: 1
         min_quantity: 2
-        base: 1
+#        base: 1
```
Comment (legalsylvain) [inline pos_pricelist/demo/pos_pricelist_demo.yml]: same here, useless line I guess.

**Comment by legalsylvain** [comment]
Hi @x620, 
Thanks for porting this important module CC AdilHoumadi  ! You marked the last commit with 'WIP' tag. is this PR still Work in progress or do you review review ?
some other comment inline.

Other points, You can take a look on the following PR that improve the V8 version :
-  #79 (CC @StefanRijnhart) 
- #71 (CC idem)
- #56 (CC @antespi)

Maybe you could git cherry-pick all / some commits done against V8, since V9 branches has been created.

Thanks again and kind regards.

File: pos_pricelist/static/src/js/models.js [hunk 3]
```diff
@@ -15,24 +15,217 @@
...
+	var screens = require('pos_pricelist.widgets');
+	var core = require('web.core');
+	var gui = require('point_of_sale.gui');
 
-    /**
+	var _t = models._t;
+    var round_pr = models.round_precision;
+    var round_di = models.round_decimals;
+
+	PosDB = PosDB.extend({
```
Comment (yelizariev) [inline pos_pricelist/static/src/js/models.js:30]: Move this block to db.js

**Comment by x620** [comment]
During running the application an errors does not appear. What to look for and what to consider in order to be sure that the application works correctly?

**Comment by flotho** [comment]
Hi @x620 @yelizariev ,

I've made some additional test in your db.
We have a pricelist : http://3143544-83-bcbce2.runbot1.odoo-community.org/web#id=3&view_type=form&model=product.pricelist&menu_id=113&action=122
And agrolait is affected to the pricelist : http://3143544-83-bcbce2.runbot1.odoo-community.org/web#id=7&view_type=form&model=res.partner&menu_id=68&action=54
Yet when selling products of the pricelist for agrolait in the POS the price si not discounted.
Could it come from the fact that the list price item is based on template instead of variant?
I also made some tests :
- First line is on variant
- second on template
- Rest as a global fucntion with discount

None of those rules are applied

**Comment by flotho** [comment]
Hi @x620 , 

Do you want us to make additionnal tests?

**Comment by x620** [comment]
Hi flotho. 
Yet only tooltips.

**Comment by flotho** [comment]
Ok, we'll wait, did you got the test use cases we made ?

regards

**Comment by x620** [comment]
Yes. Thanks

**Comment by flotho** [comment]
Thanks to you!

**Comment by x620** [comment]
Flotho, I made a few corrections. Can you to make additional tests?

**Comment by flotho** [comment]
Hi @x620 ,

We made additionnal tests.
We closed the session and restart new ones, change the pricelist, use fixed price, formula, etc... without success.

Our workflow of testing :
- Start a brand new session
- Create a new order
- Use agrolait and set the customer on each new transaction
- Choose one of the products in the pricelist (those with fixed price, percentage, etc...)

=> the price used in the order line is not the correct one

**Comment by flotho** [comment]
@x620 does the links specified here https://github.com/OCA/pos/pull/83#issuecomment-195376973 still in the correct DB?
Maybe we are not working on the correct one.

**Comment by flotho** [comment]
test updated here : http://3143822-9-0-0b7bdb.runbot2.odoo-community.org/web#id=4&view_type=form&model=product.pricelist&action=228
Customer agrolait
No changes in the pos.order.line prices : http://3143822-9-0-0b7bdb.runbot2.odoo-community.org/web?#id=1&view_type=form&model=pos.order&menu_id=557&action=783
Sorry ;-)

**Comment by flotho** [comment]
Hi @x620 don't hesitate to tell us when to test.

regards

**Comment by x620** [comment]
I think it is too early. I prefer to move in small steps. I have not yet all completed.

**Comment by x620** [comment]
Hi flotho. I corrected display price in order card and few other corrections. I think that it is possible to start additional test.

File: pos_pricelist/__openerp__.py [hunk 4]
```diff
@@ -20,9 +20,12 @@
     'name': 'POS Pricelist',
     'version': '9.0.1.2.0',
     'category': 'Point Of Sale',
+    'images': [],
     'sequence': 1,
-    'author': "Adil Houmadi @Taktik,Odoo Community Association (OCA), IT-Projects LLC, Pavel Romanchenko",
+    'author': "Adil Houmadi @Taktik,Odoo Community Association (OCA), \
+    IT-Projects LLC, Pavel Romanchenko",
     'summary': 'Pricelist for Point of sale',
+    'license': 'LGPL-3',
```
Comment (yelizariev) [inline pos_pricelist/__openerp__.py]: change license to AGPL-3

**Comment by x620** [comment]
Somebody knows that does mean this error in Travis CI test:

2016-03-17 13:21:09,180 23857 CRITICAL openerp_test openerp.service.server: Failed to initialize database `openerp_test`.
Traceback (most recent call last):
File "/home/travis/OCB-9.0/openerp/service/server.py", line 885, in preload_registries
registry = RegistryManager.new(dbname, update_module=update_module)
File "/home/travis/OCB-9.0/openerp/modules/registry.py", line 386, in new
openerp.modules.load_modules(registry._db, force_demo, status, update_module)
File "/home/travis/OCB-9.0/openerp/modules/loading.py", line 419, in load_modules
cr.execute("SELECT name FROM ir_module_module WHERE state='installed' and name in %s", (tuple(mods),))
File "/home/travis/OCB-9.0/openerp/sql_db.py", line 141, in wrapper
return f(self, _args, *_kwargs)
File "/home/travis/OCB-9.0/openerp/sql_db.py", line 220, in execute
res = self._obj.execute(query, params)
ProgrammingError: syntax error at or near ")"
LINE 1: ...FROM ir_module_module WHERE state='installed' and name in ()

How to relations the error to the pos_pricelist module?

**Comment by pedrobaeza** [comment]
It was a problem we had yesterday with OCB, but it's solved. I have relaunch the build.

**Comment by flotho** [comment]
Hi, Same thing here : http://3143940-83-a86483.runbot1.odoo-community.org/web?debug=#id=3&view_type=form&model=product.pricelist&menu_id=113&action=122
Am I forgetting something in my conf? The pricelist is attached to agrolait and is not working

**Comment by flotho** [comment]
in the console :

```
error: Some modules could not be started 
Failed modules:          ["pos_pricelist.widgets"] 
Debug:                   Object {pos_pricelist.widgets: Object}
```

File: pos_pricelist/static/src/js/widgets.js [hunk 5]
```diff
@@ -66,7 +72,7 @@ function pos_pricelist_widgets(instance, module) {
         }
     });
 
-    module.PosBaseWidget.include({
+    screens.PosBaseWidget.include({
```
Comment (yelizariev) [inline pos_pricelist/static/src/js/widgets.js]: PosBaseWidget can be reached as following:

```
 var PosBaseWidget = require('point_of_sale.BaseWidget')
```

**Comment by flotho** [comment]
@yelizariev , will @x620 push this modification soon?

**Comment by x620** [comment]
Hi, flotho. Pricelist work with Global, Product Category and Product Variant. Product doesn't work, but it seems it doesn't work in 8 too.

**Comment by flotho** [comment]
Hi, my project manager validate your work, perfect thanks

**Comment by flotho** [comment]
Hi have some issue with fiscal positions when loading the POS
It is said that I haven't got access right.
Once your module is deactivate it works,
I have only some customers that have fiscal position.
What could be the issue?
Regards

**Comment by yelizariev** [comment]
Sounds like user doesn't have access to **account.fiscal.position.tax**
https://github.com/x620/pos/blob/pos-9.0-pos_pricelist/pos_pricelist/static/src/js/models.js#L651

**Comment by legalsylvain** [comment]
hi, 
I don't think if it's related, but an access right bug of Odoo core on 'account.fiscal.position.tax' model is fixed in this V 8.0 OCA PR https://github.com/OCA/pos/pull/85
CC : @pedrobaeza 
Regards.

**Comment by flotho** [comment]
Could it be related to multi company environment?

**Comment by flotho** [comment]
Obviously it is,

I tested with 2 multi company rules for account.fiscal.position.tax and account.fiscal.position.account and it works now,
@x620 could you add those rules and fields  based on the model of @pedrobaeza 

Thanks

**Comment by pedrobaeza** [comment]
I has a similar problem and make this PR: https://github.com/OCA/pos/pull/85

**Comment by flotho** [comment]
@x620, @yelizariev is there anything special with pos_cache that could prevent the module to work?

regards

**Comment by yelizariev** [comment]
@flotho I think updates from #85 should be applied when that request is merged

**Comment by flotho** [comment]
@yelizariev , the #85 is for v8 ;-)

**Comment by flotho** [comment]
@yelizariev , @x620,

what is your position regarding my latest comment? 

Thanks

**Comment by yelizariev** [comment]
@flotho those updates have to be reviewed and accepted for 8.0 first, then we can apply it for 9.0

**Comment by flotho** [comment]
ok, I understand the process.
I'll implement it form my side as it solve the issue

**Comment by flotho** [comment]
@yelizariev,

I have another problem, locally I have no issue in .js yet the pricelist don't seems to apply. I have combinations of products / category etc...
What is the correct workflow? Close the sessions? reinstall/refresh the pricelist....
I have about 45000 products and about 60000 customers so loading is long long...

**Comment by yelizariev** [comment]
@flotho try check that js is loaded in broswer

**Comment by flotho** [comment]
@yelizariev, .js is not loaded even in the Back Office, I'll dig in and let you know

**Comment by flotho** [comment]
@yelizariev could it be conflicts with your pos-addons repository ?

**Comment by yelizariev** [comment]
@flotho there could be conflicts with any modules

**Comment by flotho** [comment]
Hi @x620 , @yelizariev 

Does the field display_price_with_taxes still usefull as Odoo now have field called iface_tax_included ?

**Comment by flotho** [comment]
Hi @x620 , @yelizariev ,

I have strange issues while loading :
odoo_v10
And more.

But it looks like there is a conflict with pos_disable_payment and multi session.

Could it come from the pos_cache ?

Any advice would be appreciated,

regards

**Comment by flotho** [comment]
Additionnal tests here :
- Desinstaling multi session and disable payment make the module work

Regarding the functions, I worked with this pricelist :
odoo_v10
- The pricelist don't work with fixed price => How can I help you to debug this one
- The pricelist works with % reductions

As you can see here :
odoo_v10

test florent is correctly used with the reduction, but not the "florent bestiole" that should be a fixed price

**Comment by yelizariev** [comment]
@flotho 

> Does the field display_price_with_taxes still usefull as Odoo now have field called iface_tax_included ?

probably, we can use  iface_tax_included.

**Comment by yelizariev** [comment]
@flotho https://github.com/OCA/pos/pull/83#issuecomment-202580863
do you reproduce this issue on runbot, that is without additional modules?

**Comment by flotho** [comment]
Hi  @yelizariev , 

Sorry for having previously said that it was fully functionnal on runbot. 
We have the same issue here regarding https://github.com/OCA/pos/pull/83#issuecomment-202580863 :
odoo_v10
here is the result :
odoo_v10

Some kind of sequence might not be used in your code as the default global seems to apply correctly with % discount.

**Comment by yelizariev** [comment]
Do you mean that module doesn't apply sequence of pricelist rules?

**Comment by flotho** [comment]
I'm wondering if it is not the problem. As you could see on my first test, fixed price are not applied but discount and global are, even on the products that have fixed price. As they are the first in the list, I suspect this kind of behaviour with sequence.
Can't it be an indication?

regards

**Comment by yelizariev** [comment]
@flotho 
It seems that 8.0 has the same behavior: price calculation is stopped after first found rule
https://github.com/OCA/pos/blob/8.0/pos_pricelist/static/src/js/models.js#L530

**Comment by legalsylvain** [comment]
https://github.com/OCA/pos/pull/85 (for v8) has been merged. could you cherry pick that commit in this PR ? https://github.com/OCA/pos/pull/85/commits/b9221e27a3bd88b1331cf3c573b50d831ff2f269

kind regards.

**Comment by yelizariev** [comment]
@x620 
could please make cherry-pick:

```
git fetch upstream
git cherry-pick b9221e27a3bd88b1331cf3c573b50d831ff2f269
git push

```

**Comment by flotho** [comment]
@yelizariev ,

I'm not sure to understand your answer regarding the sequence. I understand that once a rule is found for a product, it's unnecessary to go further, yet as you can see in my previous screenshots, "2L Evian" with fixed price is my first rule regarding the new way of sorting rules in odoo v9. So for this product, this rule should have been preferred to the default one which is global.
As you know, pricelist structure has changed, for example, here is the content of the pricelist I first displayed :

```
SELECT pricelist_id, sequence, fixed_price, create_date, price_discount, applied_on
       percent_price, date_start, company_id, product_tmpl_id,  
       price_min_margin, compute_price, base, write_date, categ_id, 
       price_surcharge, price_round, product_id, base_pricelist_id
  FROM product_pricelist_item
  WHERE pricelist_id=8;

Pricelist   seq fixed_price appliedon       
8           5                           3_global        formula
8           5   456                 0_product_variant   0.000   fixed
8           5   0                   0_product_variant   0.000   percentage

```

As you can see Odoo make its own sort and always finished and display list price with global at the end, and the sequence is now used in a second step.
My opinion is that the method used to sort https://github.com/OCA/pos/blob/9.0/pos_pricelist/static/src/js/db.js#L128 has to be rewrite a little so that the type of rule is preferred as a first key for sorting.

What do you think of this ?

**Comment by yelizariev** [comment]
@flotho ok, thanks for explanation

@x620 could you please add sorting on applied_on field so that 3_global will beat  the end

**Comment by flotho** [comment]
@yelizariev , @x260, a great thanks for this.

regards

**Comment by flotho** [comment]
@yelizariev , @x620 , I have an additionnal question regarding categories.
If you have 2 categories like :
A - Main category and B-child category which is a child of A
How do you deal with products in Categ B if you set up a pricelist rule on category A? Do you browse all products belonging to the tree of the category or directly attached on the category?
Sorting the rules by category with the lower level first could be a way of implementing this .

Thanks for your feedback!

**Comment by yelizariev** [comment]
@flotho  
for products in category B system checks is there rule for category B and if there is, such rule is applied, if not system checks the same for parent of category B and so on
https://github.com/OCA/pos/blob/8.0/pos_pricelist/static/src/js/models.js#L452

**Comment by flotho** [comment]
@yelizariev , thanks

Great News!

**Comment by flotho** [comment]
Hi all of you, personnaly I validate that the functionnal part is working.
FMO, it could be merged.

regards

**Comment by flotho** [comment]
Hi @x620 , @yelizariev , 
I have a strange issue that I didin't succeed to debug. On my localhost it works perfectly, on my preprod I have such a result :
odoo_v10

for those rules :
odoo_v10

The behaviour is as the prices in rules weren't not used. I'm up to date and working with this one : https://github.com/MindAndGo/pos

Could you lead me for debugging? 

regards

**Comment by flotho** [comment]
@yelizariev @x620 ,

I have a question. I made a test with a single pricelist in which there are some rules for a specific variant regarding to the quantity. If I select a customer, the correct prices are displayed, if I don't choos the pricelist, the rules are recognised but not the prices.
It looks like a bug to me, shouldn't the price be updated with each rules even if no customer is used?

regards

**Comment by flotho** [comment]
Regarding the point of the rules that are not correctly applied, I think I've found what's going on. If you add a price to a product, it's added by default with a rule on template.
If you change it later and want to set it up on a variant, the information on template still exists :
odoo_v10
As you can see, some rules keeps together variant_id AND template id. So my opinion is to not exclude based on template rules IF the variant one is selected and also if applied on is set to variant.
Could it be reviewed?

**Comment by yelizariev** [comment]
@flotho as I understand, you asking to change current behavior https://github.com/OCA/pos/pull/83#issuecomment-202774766

**Comment by flotho** [comment]
@yelizariev , @x620 ,

regarding the default behaviour on pricelist when no partner is chosen, could it come from the switch https://github.com/OCA/pos/pull/83/files#diff-747f62729cafe771e948f5cfeb9aff46R456 and the different cases used?

**Comment by yelizariev** [comment]
@flotho this code could be the reason.

```
                if (!partner || rule.pricelist_id[1] != partner.property_product_pricelist[1]) {
                        continue
            }
```

@x620 could you explain why do you do such check? Doesn't mean, that pricelist is not applied if there is no customer choosen?
https://github.com/x620/pos/commit/95173e245cb9ce5b2acda616586c322e162093c7

**Comment by flotho** [comment]
@yelizariev , thanks for your answer 

> @flotho as I understand, you asking to change current behavior #83 (comment)

Not really, my point is not to replace this behaviour but to make it stronger. I explained. 
As you see on my latest screenshot, the pricelist rule is applied_on variant YET odoo (not your code) let the information about the template in the rule.
So my point is that somewhere in your code you are giving the priority to a control on the field template_id instead of the applied_on field .
So the previous behaviour should not been replaced but enhanced so that the logic could be respected even if Odoo don't update all the fields correctly.

Tell me if I'm miss something in my explanations

regards

**Comment by yelizariev** [comment]
Do you mean to give rules on Template more priorotiy that on Variant? I don't think that it's a good idea.

```
    'applied_on': fields.selection([('3_global', 'Global'),('2_product_category', ' Product Category'), ('1_product', 'Product'), ('0_product_variant', 'Product Variant')], string="Apply On", required=True,
```

**Comment by flotho** [comment]
@yelizariev , not at all, the applied on is ok.
My point is that when a rule keeps the template info (bug from odoo ?) even if it's specified as applied_on variant, the price computation failedand keep the default price.
That's the point of https://github.com/OCA/pos/pull/83#issuecomment-208223616 .

And it's mostly related to this part of code : https://github.com/OCA/pos/pull/83/files#diff-747f62729cafe771e948f5cfeb9aff46R417

**Comment by flotho** [comment]
Hi @yelizariev, to summarize we still have :
- 1 bug related to https://github.com/OCA/pos/pull/83#issuecomment-208252262
- 1 possible bug when there are rules based on product (1_product) seems to not be applied!

**Comment by flotho** [comment]
@x620 , @yelizariev ,

Any feedback on those 2 little bugs? regards

**Comment by yelizariev** [comment]
@flotho  updates are coming soon

**Comment by flotho** [comment]
@yelizariev , thanks for the feedback,

**Comment by flotho** [comment]
@x620 , @yelizariev ,

Do you want me to test now?
regards

**Comment by yelizariev** [comment]
@flotho yes, try now

**Comment by flotho** [comment]
Hi  @yelizariev , @x620 ,

i've made additionnal tests, here are the results :
- When no customer is selected the default pricelist is applied with correct quantities and amount => OK
- When rules are defined for template => KO:
  odoo_v10
  it still not working:
  odoo_v10

tested here : http://3145222-83-f7812c.runbot2.odoo-community.org/web#id=2&view_type=form&model=product.pricelist&menu_id=113&action=122

Hope it could help

File: pos_pricelist/static/src/js/models.js [hunk 6]
```diff
@@ -410,18 +416,20 @@ function pos_pricelist_models(instance, module) {
             var items = [], i, len;
             for (i = 0, len = db.pricelist_item_sorted.length; i < len; i++) {
                 var item = db.pricelist_item_sorted[i];
-                if ((item.product_id === false
-                    || item.product_id[0] === product.id) &&
-                    (item.categ_id === false
-                    || categ_ids.indexOf(item.categ_id[0]) !== -1) &&
-                    (item.price_version_id[0] === version.id)) {
+                if (
+                    (item.product_tmpl_id === false || item.product_tmpl_id[0] === product.id)
```
Comment (yelizariev) [inline pos_pricelist/static/src/js/models.js]: I believe it must be  like this:

```
  (item.product_tmpl_id === false || item.product_tmpl_id[0] === product.product_tmpl_id)
```

**Comment by yelizariev** [comment]
@flotho try again

**Comment by flotho** [comment]
Hi @yelizariev ,

It's better, we're almost done.
Here is y conf :
Default price list
odoo_v10
Specific price lis associated to Agrolait
odoo_v10

Here are the results of my tests  :

1 - Without any customer and a  => almost OK, why does the evian template display 2 lines?
odoo_v10

2 - Start a new order with customer rules displayed seems to be uncomplete (it behaves correctly regarding to the quantity) :
odoo_v10

In the end, the function seems to be Ok but it looks like there is a regression on the displayed popups.

**Comment by x620** [comment]
Hi, @flotho 
In both cases tooltips also work and in 8.
odoo8_a

odoo8_b

But in the second case, it can be easily eliminated by adding a row in the pricelist: All products 1x 0% discount. Then we get:

odoo8_c

**Comment by flotho** [comment]
Ok, it just looks weird to me but still functionnal.
So we could say that it's Ok

**Comment by flotho** [comment]
Hi @yelizariev , @x620 ,

I have a strange conflict when  I use this module with others. The product.product model is not found while loaded so that some fields are missing in the model.
It looks like using initialize.apply mess with another part. What is the correct way to push fileds for product.product in other modules so that there won't be conflicts with yours?

**Comment by flotho** [comment]
Hi @yelizariev , @x620 ,

i'm not sure that it is te most proper way to deal with my issue but I change a little bit your syntax https://github.com/MindAndGo/pos/commit/9c79f80ccdecaf6d3df9cc8bd6267a9be2f787c0 and it works for me. The categ_id are always loaded.
I'm sure there is a conflicts with other modules loading the product.product model but I can't find what's the proper way to deal with that so, I use this patch.
Your feedback would be appreciated.

regards

**Comment by flotho** [comment]
Hi @yelizariev , @x620 ,

Last bug really touchy I think. I tested on the latest runbot.
I have the main pricelist with quantities :
odoo_v10

Then another one pricelist that depends on the first one :
odoo_v10

Once you select your customer, your module succeed to identify that there are many lines but the price is not correctly used
odoo_v10

tested there : http://3146099-83-7c28e6.runbot1.odoo-community.org/web#id=2&view_type=form&model=product.pricelist&menu_id=113&action=122

**Comment by yelizariev** [comment]
> It looks like using initialize.apply mess with another part. What is the correct way to push fileds for product.product in other modules so that there won't be conflicts with yours?

It should be inheritance instead of overwriting, because overwriting reject all previous inheritances
https://odoo-development.readthedocs.org/en/latest/dev/pos/add_field.html

**Comment by flotho** [comment]
Hi @yelizariev ,

> It should be inheritance instead of overwriting, because overwriting reject all previous inheritances
> https://odoo-development.readthedocs.org/en/latest/dev/pos/add_field.html

I tried this one and have some problems after... I'll regive it a try

**Comment by x620** [comment]
Hi @flotho 
You can now test.

**Comment by StefanRijnhart** [comment]
It seems you kept all the changes that are intended to support fiscal positions in the POS. This was a rather 'hidden' functionality of this module, and it was not completely functional. As the point_of_sale module in Odoo 9.0 has native support for fiscal positions, you should weed out all these changes.

**Comment by StefanRijnhart** [comment]
If you are nice, you migrate the cross table that is used for pos.order.line's tax_ids in the 8.0 version of this module to the cross table that is used in 9.0's point_of_sale for the same field in a migration script.

**Comment by flotho** [comment]
Hi @x620 ,

i'll test.
Could you also deal with this part : https://github.com/OCA/pos/pull/83#issuecomment-215033132 and integrate it in the next commit?

@yelizariev , what about the proposal of @StefanRijnhart , could it be included in my latest financial support?

**Comment by yelizariev** [comment]
@flotho 
pos_pricelist does make inheritance and not overwriting https://github.com/x620/pos/blob/pos-9.0-pos_pricelist/pos_pricelist/static/src/js/models.js#L703-L711 
I meant that other modules can do overwriting that broke system

**Comment by yelizariev** [comment]
@StefanRijnhart is it correct, that all files in models/ folder are related to fiscal position and can be removed?

**Comment by flotho** [comment]
@yelizariev 
I'm no more sure to understand because you mentionned https://odoo-development.readthedocs.io/en/latest/dev/pos/add_field.html . Does that means that we have to .call or to .apply?

Anyway, thanks for your support

**Comment by yelizariev** [comment]
@flotho it's not a big difference between .call and .apply

You can find an example of wrong way to add fields here - https://github.com/yelizariev/pos-addons/blob/8.0/tg_pos_enhanced/static/src/js/tg_pos.js#L17
-- it comletly overwrites load_server_data function

**Comment by StefanRijnhart** [comment]
@yelizariev looks like it, except for the definition of the display_price_with_taxes field, which is another side feature of this module.

**Comment by flotho** [comment]
@x620 , @yelizariev 

From my point of view, it works !

**Comment by phoenixsampras** [comment]
Awesome work!! please keep it on!! this is so necessary for supermarkets...

**Comment by woepwoep** [comment]
pls connect with me re supermarkets
i am in the food cooperative industry, helping to build a movement here in
the Netherlands.
contact details below
thx, Ron
web: www.voedsel.coop
email: info@voedsel.coop
phone: +31 629 584 582

---

http://about.me/woepwoep

2016-05-20 5:01 GMT+02:00 phoenixsampras notifications@github.com:

**Comment by flotho** [comment]
@yelizariev ,

I have a remark regarding performances. I think it becomes useless to add supplierinfo in the POS.
Actually, the default behaviour of pricelist don't use supplierinfo anymore as way of computing prices.
What is your point of view?

regards

**Comment by yelizariev** [comment]
I suggest to try merge this PR to new branch, say 9.0-pos_pricelist. This could simplify contributing by other developers. If someone find issue and find a way to fix, one will be able to create new PR instead of asking @x620 and wait when he update his PR.

See discussion on twitter: https://twitter.com/jgrandguillaume/status/735738547118825475

The same approach could be applied to other PR with `needs review` status

**Comment by gaelTorrecillas** [comment]
hello,

@yelizariev in your models.js, the function models.PosModel.extend's initialize returns nothing, why?
When I have to overload this function, I put "return _super_posmodel.initialize.call(this, session, attributes);" 
I understand that you used .apply but we don't return _super_posmodel.initialize.apply(this, arguments) ?

**Comment by gaelTorrecillas** [comment]
thanks x620 for your job.

I have other question, in
https://github.com/x620/pos/blob/pos-9.0-pos_pricelist/pos_pricelist/static/src/js/models.js#L262

the function "can_be_merged_with", why the inherit? because, if the products don't have the same id, the core returns false, and if the products' price aren't the same, the core returns false.
do we want save the inherit or if we don't save it, the point of sale will not have the problem?

**Comment by dcoopennet** [comment]
@x620 Is this PR going to be merged ? We need this PORT for one of our customers.
I can review it if you want.

**Comment by rafaelbn** [comment]
Hi @x620 , thanks a lot for this PR, thanks all for revisions. This PR is amazing! 😄 

Sorry for the late.

Please @x620 could you rebase to check again what's going on with travis?

Also it could be useful is you check this FIX: https://github.com/OCA/pos/pull/100 made by @llacroix in v8

Thanks again!

**Comment by rafaelbn** [comment]
Hi again, as commented by @yelizariev in https://github.com/OCA/pos/pull/83#issuecomment-223221180 after rebase we will check travis and runbot but we can merge if all is green. cc @x620 @legalsylvain

**Comment by yelizariev** [comment]
@rafaelbn 
Pavel (@x620) is out of business and would not make updates. So, it's probably better to create new PR with rebasing and other updates. Can you do it yourself?

**Comment by rafaelbn** [comment]
I cannot myself, but I could ask in my team. In this moment we don't need this but I wish we will. If someone is interested in POS in v9 then could do it. Let wait sometime then.

**Comment by rafaelbn** [comment]
Hi all! Is anyone interested in this PR which is a migration from v8 to v9 of `pos_pricelist`. If not I will close this PR. Thanks!

**Comment by pedrobaeza** [comment]
As a side note, Odoo v11 include this functionality in core.

**Comment by pedrobaeza** [comment]
Closing this in favor of #186

**Pull request closed without merging**